api/types/container/hostconfig.go

```diff
@@ -440,6 +441,17 @@ type HostConfig struct {
 // of the returned, including checking if the value is empty, should be handled
 // by the caller.
 func containerID(val string) (idOrName string, ok bool) {
-	k, idOrName, ok := strings.Cut(val, ":")
-	return idOrName, ok && k == "container"
+	k, v, hasSep := strings.Cut(val, ":")
+	ok = hasSep && k == "container"
+	if !ok {
+		return "", ok
+	}
```

Comment:
```suggestion
	if !hasSep || k != "container" {
		return "", false
	}
```

Reply:
Heh, I went a bit back-and-forth on what was easiest to read. Let me change it to your suggestion